src/structures/CommandInteraction.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+const APIMessage = require('./APIMessage');
+const Interaction = require('./Interaction');
+const WebhookClient = require('../client/WebhookClient');
+const { ApplicationCommandOptionTypes, InteractionResponseTypes } = require('../util/Constants');
+const MessageFlags = require('../util/MessageFlags');
+
+/**
+ * Represents a command interaction.
+ * @extends {Interaction}
+ */
+class CommandInteraction extends Interaction {
```

Comment:
Personally I don't quite like how command-related info is split across three keys.
Instead of having `commandID`, `commandName` and `options`, why not group that under a key like `commandInfo`?

Reply:
The current naming fits better with other conventions in the library.

Besides, I would argue that most of the time you as a user, you would be using the `CommandInteraction#command` getter, which internally uses the `CommandInteraction#commandID` to return the full `ApplicationCommand` object

Reply:
My thought was that the key name `options` would seem a bit too generic, but if that goes with the conventions, then that's alright.

I assumed `Interactions` would also reflect other types of interactions, but if that's only the slash commands, that would not be an issue.

Reply:
While I agree that `commandID` and `commandName` *could* be grouped together, I don't think it would make much sense, considering that you have the `command` getter in case you have fetched the command(s) before. As for `options`, this should stay on the top level, otherwise it could be confused with `ApplicationCommand#options` which has a different structure than `CommandInteraction#options` does.